src/main/java/com/cloudbees/plugins/credentials/CredentialsDescriptor.java

```diff
@@ -192,4 +257,339 @@ public String getIconClassName() {
     public boolean isApplicable(CredentialsProvider provider) {
         return true;
     }
+
+    /**
+     * In some cases the nearest {@link AncestorInPath} {@link ModelObject} is one of the Credentials plugins wrapper
+     * classes.
+     * This helper method unwraps those to return the correct context.
+     *
+     * @param context the context (wrapped or unwrapped).
+     * @return the unwrapped context.
+     * @since 2.1.5
+     */
+    @NonNull
+    public static ModelObject unwrapContext(@NonNull ModelObject context) {
+        if (context instanceof CredentialsSelectHelper.WrappedCredentialsStore) {
```

Comment:
same `getStore()` method on all classes. Seems like an interface could clean this up a bit and lend to easier future expansion.